src/packages/mikroorm/src/provider/provider.ts

```diff
@@ -462,10 +462,10 @@ export class MikroBackendProvider<D> implements BackendProvider<D> {
 		let queryFilter: any = { [relatedField]: { $in: relatedFieldIds } };
 
 		if (filter) {
+			// JSON.parse(JSON.stringify()) is needed. See https://exogee.atlassian.net/browse/EXOGW-419
+			const gqlToMikroFilter = JSON.parse(JSON.stringify([gqlToMikro(filter, this.getDbType())]));
```

Comment:
Just curious what sort of keys and values are you trying to remove here?

Reply:
We are not trying to remove any properties from the queryFilter object. However, Mikro-ORM internally checks if the object is a POJO or if it was constructed from a class or has any other inherited methods that are not just plain properties. Mikro-ORM will treat filters as plain objects and non-plain objects differently, and the resulting SQL Query was wrong.
The `filter` object coming as a param to this `findByRelatedId` method is not a plain object, it comes directly from GraphQL land, so the resulting SQL was not as we expected. Parse-stringifying this object will give as a POJO which resulting in Mikro-ORM giving us the correct SQL.